airbyte-integrations/connectors/source-facebook-marketing/source_facebook_marketing/source.py

```diff
@@ -183,6 +184,12 @@ def get_custom_insights_streams(self, api: API, config: ConnectorConfig) -> List
         """return custom insights streams"""
         streams = []
         for insight in config.custom_insights or []:
+            insight_fields = set(insight.fields)
+            if insight_fields.intersection(UNSUPPORTED_FIELDS):
+                # https://github.com/airbytehq/oncall/issues/1137
+                logger.info(f"Following fields were removed from the Custom{insight.name} fields list due to error"
+                            f" on Facebook side: {insight_fields.intersection(UNSUPPORTED_FIELDS)}")
+                insight_fields -= UNSUPPORTED_FIELDS
```

Comment:
can we validate presence of these fields in the `check_connection` to prevent user from providing an invalid config and sooner highlighting the issue?

Reply:
`check_connection` uses this function, do you want to raise an error here instead of quiet deleting?

Reply:
Yes, I think it's a better UX

Reply:
updated